### PR TITLE
add html proofer check

### PR DESCRIPTION
**Pull request by xu-cheng**
`html proofer` is a small tool to [validated the sources in web page](https://github.com/gjtorikian/html-proofer#whats-tested). We can use it and Travis-CI to check http://brew.sh.

You can see example ci result [here](https://travis-ci.org/xu-cheng/homebrew.github.io/builds/52117448). `html proofer` spots the following errors:

```
- ./_site/css/fonts/chunkfive/demo.html
  *  internally linking to stylesheet.css, which does not exist (line 9)
- ./_site/index_es.html
  *  External link http://catalandr.es/ failed: 404 No error
```

I'm not a front-end web developer. So I leave other people to fix the problem above.

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -0,0 +1,20 @@
+require "rake"
+
+task :default => :build
+
+# rake build
+desc "Build the site"
+task :build do
+  shell_args = %w[bundle exec jekyll build]
+  sh *shell_args
```
Comment (mikemcquaid) [inline Rakefile]: Probably worth just inlining this argument. Does Jekyll have any Rake helpers we can use?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+language: ruby
+sudo: false
+cache: bundler
+rvm:
+ - 2.2
```
Comment (mikemcquaid) [inline .travis.yml]: Maybe set this to 1.8 so we can allow people on 10.8 to test this easier?

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+language: ruby
+sudo: false
```
Comment (mikemcquaid) [inline .travis.yml]: Why do we need this?

File: Rakefile (hunk 1 above)
Comment (xu-cheng) [inline Rakefile]: Yes, I should inline it. I copy it from the Rakefile of my own website. So forget to do so. 

> Does Jekyll have any Rake helpers we can use?

Not that I am aware of.

File: .travis.yml (hunk 3 above)
Comment (xu-cheng) [inline .travis.yml]: If will make ci run in the docker-based container. It's good for performance.

Comment (mikemcquaid) [inline .travis.yml]: Interesting.

File: Rakefile [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+task :default => :build
+
+# rake build
+desc "Build the site"
+task :build do
+  sh "bundle exec jekyll build"
+end
+
+# rake test
```
Comment (mikemcquaid) [inline Rakefile]: Can probably kill the comments.

File: Rakefile [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+desc "Html Proofer"
```
Comment (mikemcquaid) [inline Rakefile]: Might be worth expanding this description a bit.

**Comment by mikemcquaid** [comment]
> - ./_site/css/fonts/chunkfive/demo.html
>   *  internally linking to stylesheet.css, which does not exist (line 9)

Think we can just remove this page altogether.

> - ./_site/index_es.html
>   *  External link http://catalandr.es/ failed: 404 No error

Can you remove this link? Thanks.

File: .travis.yml (hunk 2 above)
Comment (xu-cheng) [inline .travis.yml]: 1.8 is a dead end. `i18n` requires at least 1.9. So I have set it to 1.9.

**Comment by xu-cheng** [comment]
Now CI is [green](https://travis-ci.org/xu-cheng/homebrew.github.io/builds/52128342)

**Comment by mikemcquaid** [comment]
Can you force-push again? I've enabled Travis so hopefully it'll build this after that.

**Comment by xu-cheng** [comment]
Done.

**Comment by xu-cheng** [comment]
Green. :rocket:

**Comment by mikemcquaid** [comment]
Hooray.

**Comment by xu-cheng** [comment]
Also don't forget to add CI Badge in README.

**Comment by mikemcquaid** [comment]
@xu-cheng Feel free :wink:

**Comment by xu-cheng** [comment]
https://github.com/Homebrew/homebrew.github.io/commit/b39b8c9f9d8dee2cd5c82db8b38dd356935f9fc8

**Pull request merged**